lizard_auth_client/decorators.py

```diff
@@ -0,0 +1,55 @@
+# (c) Nelen & Schuurmans.  GPL licensed, see LICENSE.rst.
+# -*- coding: utf-8 -*-
+
+"""View decorators."""
+
+# Python 3 is coming
+from __future__ import unicode_literals
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import division
+
+from functools import wraps
+from django.contrib.auth.views import redirect_to_login
+from django.conf import settings
+from django.utils.decorators import available_attrs
+from django.utils.encoding import force_str
+from django.utils.six.moves.urllib.parse import urlparse
+
+SESSION_ATTEMPT_MADE = 'LIZARD_AUTH_CLIENT_LOGIN_ATTEMPT_MADE'
+
+
+def attempt_auto_login(view):
+    """Attempt to login an unauthenticated user using the SSO server.
+
+    Contrary to login_required, this does not give an error or redirect
+    to a login page if the user isn't logged in on the SSO server yet,
+    but will then simply continue with an unauthenticated user.
+
+    This attempt is only made once per session. The session field
+    'LIZARD_AUTH_CLIENT_LOGIN_ATTEMPT_MADE' will be set, and subsequent
+    calls will just continue on to the view with the user as currently
+    set in the session.
+    """
+
+    @wraps(view, assigned=available_attrs(view))
+    def wrapped_view(request, *args, **kwargs):
+        if (SESSION_ATTEMPT_MADE in request.session or
+                request.user.is_authenticated()):
+            return view(request, *args, **kwargs)
+
+        request.session[SESSION_ATTEMPT_MADE] = True
```

Comment:
Dit is wel iets dat lang in de sessie blijft staan. Gaat dat goed? Kan er niet beter een expiration date bij? Zoiets:

`request.session[SESSION_ATTEMPT_LAST_MADE] = datetime.datetime.now()`